Kinvey/Kinvey/RequestFactory.swift

```diff
@@ -10,56 +10,97 @@ import Foundation
+    func buildAppDataFindByQuery(collectionName: String, query: Query, options: Options?) -> HttpRequest
+    func buildAppDataCountByQuery(collectionName: String, query: Query?, options: Options?) -> HttpRequest
+    func buildAppDataGroup(collectionName: String, keys: [String], initialObject: [String : Any], reduceJSFunction: String, condition: NSPredicate?, options: Options?) -> HttpRequest
+    func buildAppDataSave<T: Persistable>(_ persistable: T, options: Options?) -> HttpRequest
+    func buildAppDataRemoveByQuery(collectionName: String, query: Query, options: Options?) -> HttpRequest
+    func buildAppDataRemoveById(collectionName: String, objectId: String, options: Options?) -> HttpRequest
+    func buildAppDataSubscribe(collectionName: String, deviceId: String, options: Options?) -> HttpRequest
+    func buildAppDataUnSubscribe(collectionName: String, deviceId: String, options: Options?) -> HttpRequest
+    
+    func buildPushRegisterDevice(_ deviceToken: Data, options: Options?) -> HttpRequest
+    func buildPushUnRegisterDevice(_ deviceToken: Data, options: Options?) -> HttpRequest
+    
+    func buildBlobUploadFile(_ file: File, options: Options?) -> HttpRequest
+    func buildBlobDownloadFile(_ file: File, options: Options?) -> HttpRequest
+    func buildBlobDeleteFile(_ file: File, options: Options?) -> HttpRequest
+    func buildBlobQueryFile(_ query: Query, options: Options?) -> HttpRequest
+    
+    func buildCustomEndpoint(_ name: String, options: Options?) -> HttpRequest
+    
+    func buildOAuthToken(redirectURI: URL, code: String, options: Options?) -> HttpRequest
+    
+    func buildOAuthGrantAuth(redirectURI: URL, options: Options?) -> HttpRequest
+    func buildOAuthGrantAuthenticate(redirectURI: URL, tempLoginUri: URL, username: String, password: String, options: Options?) -> HttpRequest
+    func buildOAuthGrantRefreshToken(refreshToken: String, options: Options?) -> HttpRequest
+    
+    func buildLiveStreamGrantAccess(streamName: String, userId: String, acl: LiveStreamAcl, options: Options?) -> HttpRequest
+    func buildLiveStreamPublish(streamName: String, userId: String, options: Options?) -> HttpRequest
+    func buildLiveStreamSubscribe(streamName: String, userId: String, deviceId: String, options: Options?) -> HttpRequest
+    func buildLiveStreamUnsubscribe(streamName: String, userId: String, deviceId: String, options: Options?) -> HttpRequest
+    
+}
+
+//Allow to set a per request
+public struct Options {
+    
+    public var client: Client?
+    public var clientId: String?
+    public var ttl: TTL?
+    public var deltaSet: Bool?
+    public var readPolicy: ReadPolicy?
+    public var writePolicy: WritePolicy?
+    public var timeout: TimeInterval?
+    
+    /// App version for this client instance.
+    public var clientAppVersion: String?
+    
+    /// Custom request properties for this client instance.
+    public var customRequestProperties: [String : Any]?
+    
+    public init(
```

Comment:
@heyzooi Should there be an `init` with no parameters? Since all properties are public, maybe it makes sense to just `init()` and then set the ones you care about?